docs: update readme to include debugger (#46597)
ray-project/ray: README.rst

- `Actors`_: Stateful worker processes created in the cluster.
- `Objects`_: Immutable values accessible across the cluster.

Learn more about Monitoring and Debugging:

> - Debug Ray applications with `Ray Distributed Debugger <https://docs.ray.io/en/latest/ray-observability/ray-distributed-debugger.html>`__.

Reviewer: ```suggestion
- Debug Ray applications with the `Ray Distributed Debugger <https://docs.ray.io/en/latest/ray-observability/ray-distributed-debugger.html>`__.
```

> - Monitor Ray applications and clusters with `Ray dashboard <https://docs.ray.io/en/latest/ray-core/ray-dashboard.html>`__.

Reviewer: ```suggestion
- Monitor Ray apps and clusters with the `Ray Dashboard <https://docs.ray.io/en/latest/ray-core/ray-dashboard.html>`__.
```

> - Debug Ray applications with the `Ray Distributed Debugger <https://docs.ray.io/en/latest/ray-observability/ray-distributed-debugger.html>`__.

Author: ```suggestion
- Debug Ray apps with the `Ray Distributed Debugger <https://docs.ray.io/en/latest/ray-observability/ray-distributed-debugger.html>`__.
```
